plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMinioConnectorTest.java

```diff
@@ -0,0 +1,234 @@
+    private HiveMinioDataLake hiveMinioDataLake;
+
+    public BaseIcebergMinioConnectorTest(IcebergFileFormat format)
+    {
+        super(format);
+        this.bucketName = "test-iceberg-minio-integration-test-" + randomTableSuffix();
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        this.hiveMinioDataLake = closeAfterClass(new HiveMinioDataLake(bucketName, ImmutableMap.of()));
+        this.hiveMinioDataLake.start();
+        return createQueryRunner(
+                ImmutableMap.<String, String>builder()
+                        .put("iceberg.catalog.type", "HIVE_METASTORE")
+                        .put("hive.metastore.uri", "thrift://" + hiveMinioDataLake.getHiveHadoop().getHiveMetastoreEndpoint())
+                        .put("hive.s3.aws-access-key", ACCESS_KEY)
+                        .put("hive.s3.aws-secret-key", SECRET_KEY)
+                        .put("hive.s3.endpoint", "http://" + hiveMinioDataLake.getMinio().getMinioApiEndpoint())
+                        .put("hive.s3.path-style-access", "true")
+                        .put("hive.s3.streaming.part-size", "5MB")
+                        .build());
+    }
+
+    @Override
+    protected SchemaInitializer.Builder createSchemaInitializer(String schemaName)
+    {
+        return super.createSchemaInitializer(schemaName)
+                .withSchemaProperties(ImmutableMap.of("location", "'s3://" + bucketName + "/" + schemaName + "'"));
+    }
+
+    @Override
+    protected String createSchemaSql(String schemaName)
+    {
+        return "CREATE SCHEMA IF NOT EXISTS " + schemaName + " WITH (location = 's3://" + bucketName + "/" + schemaName + "')";
+    }
+
+    @Override
+    protected Long calculateFileSystemFileSize(String filePath)
+    {
+        return this.hiveMinioDataLake.getS3Client()
+                .getObject(bucketName, filePath.replace("s3://" + bucketName + "/", ""))
+                .getObjectMetadata()
+                .getContentLength();
+    }
+
+    @Override
+    protected BufferedInputStream createDataFileStreamForFileSystem(String filePath)
+    {
+        return new BufferedInputStream(this.hiveMinioDataLake.getS3Client()
+                .getObject(bucketName, filePath.replace("s3://" + bucketName + "/", "")).getObjectContent());
+    }
+
+    @Override
+    protected List<String> getAllDataFilesFromTableDirectory(String tableName)
+    {
+        String tableLocation = getSession().getSchema().orElseThrow() + "/" + tableName + "/data";
+        return this.hiveMinioDataLake.getS3Client().listObjects(bucketName, tableLocation)
+                .getObjectSummaries()
+                .stream()
+                .map(object -> "s3://" + bucketName + "/" + object.getKey())
+                .filter(object -> !object.endsWith(".crc"))
+                .collect(toImmutableList());
+    }
+
+    @Test
+    public void testRepartitionDataOnCtasPartitionComplex()
```

Comment:
it's not override. is it a copy?
what makes this minio-specific?

Reply:
It's explained in comments on override of `repartitioningDataProvider` .
To make it more clear. I've refactored now to:
- extract const with partition definitions.
- filter default test data in `repartitioningDataProvider` override
- moved those test cases below `repartitioningDataProvider` override to give it logical order